### DISCUSSION TITLE
Can the DPLR model simulate external electric field interactions?

### DISCUSSION THREAD

**Discussion by sancylb** [deepmodeling/deepmd-kit#2136] [Q&A]
Hi all, the DPLR model uses wannier centroids to include long-range electrostatic interactions. So how the DPLR model can respond to external electric field interactions? If the wannier centroids can react to external electric fields, how is it implemented? Should add new training data set with different external electric fields employed in DFT calculations? Thanks!

**Comment by wanghan-iapcm** [comment]
You may find a detailed discussion on the topic in Sect. VI of [the DPLR paper](https://aip.scitation.org/doi/10.1063/5.0083669).

In short, it is straightforward to include the effect of the external field in the current DPLR model by Eq. 34. Notice that the polarization P({R}) is given by the deep dipole model, which describes the magnitude of molecular dipole as a function of the molecular environment (the coordinates of all nearby atoms). The polarization of the system due to a small external field (in the regime Kubo's fluctuation dissipation theorem holds) can be studied by MD simulations using 34 as the PES. For larger external fields, higher order of response should be considered, like Eq.35, but this formula has not been implemented in deepmd-kit yet.

**Comment by sancylb** [comment]
Thanks for your quick reply!@wanghan-iapcm Hopefully it will be implemented in deepmd-kit soon!
